core/account-id/src/lib.rs

```diff
@@ -64,6 +64,24 @@ pub use errors::{ParseAccountError, ParseErrorKind};
 #[derive(Eq, Ord, Hash, Clone, Debug, PartialEq, PartialOrd)]
 pub struct AccountId(Box<str>);
 
+#[derive(PartialEq)]
+pub enum AccountType {
+    NamedAccount,
+    NearImplicitAccount,
+    EthImplicitAccount,
+}
+
+impl AccountType {
+    pub fn is_implicit(&self) -> bool {
+        match &self {
+            Self::NearImplicitAccount => true,
+            // TODO change to true later, see https://github.com/near/nearcore/issues/10018
```

Comment:
nit: Can you add the project name to the TODO - e.g. TODO(eth-implicit) 
This way at the end of the project you can just grep for this string and make sure all TODOs are closed.